File: Dw.ClickHouse/Dtos/Condition/ChConditionDto.cs
using Ch.Models.Dtos.Output;
using Ch.Models.Enums;
using Dw.ClickHouse.Dtos.Output;

namespace Ch.Models.Dtos.Condition
{
    public class ChConditionDto
    {
        public Conjuction? Conjuction { get; set; }
        public bool HasOpeningBracket { get; set; } = false;
        public ChOutputDto SubqueryColumn { get; set; }
        public ChColumnDto Column { get; set; }
        public Operator Operator { get; set; } = Operator.EqualTo;
        public string ExpectedResult { get; set; }
        public List<ChConditionContainDto> Contains { get; set; } = new List<ChConditionContainDto>();
        public bool HasClosingBracket { get; set; } = false;
    }
}

File: WebApplication/src/app-clickhouse/dtos/condition/ch-condition.dto.ts
import { Conjuction } from "../../enums/conjuction.enum";
import { Operator } from "../../enums/operator.enum";
import { ChColumnDto } from "../output/ch-column.dto";
import { ChOutputDto } from "../output/ch-output.dto";
import { ChConditionContainDto } from "./ch-condition-contain.dto";

export class ChConditionDto {
    conjuction: Conjuction;
    hasOpeningBracket = false;
    subqueryColumn: ChOutputDto;
    column: ChColumnDto;
    operator = Operator.equalTo;
    expectedResult: string;
    hasClosingBracket = false;

    contains: ChConditionContainDto[] = [];
}